tests/specs/helpers/uploadedFileErrorChecks.test.js

```diff
@@ -1,74 +1,124 @@
+            originalname: 'file3.pdf',
+            mimetype: 'image/png',
+        };
+        checkTypeSizeAndDuplication(
+            requestStub(),
+            newUploadedFile,
+            callbackSpy
+        );
+        expect(callbackSpy.calledWith(null, false)).to.be.true;
+    });
 
-  it("does not upload the file if it has already been uploaded", () => {
-    const previouslyUploadedFiles = [
-      {
-        filename: "file3.pdf",
-        mimetype: "application/pdf",
+    it('does not upload the file if it has already been uploaded', () => {
+        const previouslyUploadedFiles = [
+            {
+                filename: 'file3.pdf',
+                mimetype: 'application/pdf',
+            },
+        ];
+        const newUploadedFile = {
+            originalname: 'file3.pdf',
+            mimetype: 'application/pdf',
+        };
+        checkTypeSizeAndDuplication(
+            requestStub(previouslyUploadedFiles),
+            newUploadedFile,
+            callbackSpy
+        );
+        expect(callbackSpy.calledWith(null, false)).to.be.true;
+    });
 
-      },
-    ];
-    const newUploadedFile = {
-      originalname: "file3.pdf",
-      mimetype: "application/pdf",
-    };
-    checkTypeSizeAndDuplication(
-      requestStub(previouslyUploadedFiles),
-      newUploadedFile,
-      callbackSpy
-    );
-    expect(callbackSpy.calledWith(null, false)).to.be.true;
-  });
+    it('delete file and send error message if it is too large', () => {
+        // when
+        const unlinkFile = sandbox.stub(fs, 'unlink').callsFake(() => null);
+        const reqStub = {
+            files: [
+                {
+                    size: 210_000_000,
+                    originalname: 'file_1.pdf',
+                },
+                {
+                    size: 210_000_000,
+                    originalname: 'file_2.pdf',
+                },
+            ],
+            session: {
+                eApp: {
+                    uploadedFileData: [
+                        {
+                            filename: 'file_1.pdf',
+                            storageName: 'file_1.pdf',
+                        },
+                        {
+                            filename: 'file_2.pdf',
+                            storageName: 'file_2.pdf',
+                        },
+                    ],
+                    uploadMessages: {
+                        errors: [],
+                    },
+                },
+            },
+            _sails: {
+                config: {
+                    eAppS3Vals: {
+                        s3_bucket: 'leg-demo-upload-d4szp8',
+                    },
+                },
+            },
+        };
+        displayErrorAndRemoveLargeFiles(reqStub);
+
+        // then
+        expect(reqStub.session.eApp.uploadedFileData.length).to.equal(0);
+        expect(reqStub.session.eApp.uploadMessages.errors.length).to.equal(2);
+        expect(unlinkFile.callCount).to.equal(2);
+    });
 });
```

Comment:
Might be worth another test to ensure only the oversized files are removed from the set of upload files, or in other terms, if only one file is too large then the other files are uploaded successfully. Apologise if I've mis-understood the situation.